BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -971,31 +854,11 @@ public int WhiteBalance
             set
             {
                 CheckOptionSupported(RealSense2API.Option.WHITE_BALANCE, WhiteBalanceDesc.Name, RealSense2API.SensorName.COLOR);
-
-                RealSense2API.QueryOptionInfo(
-                    _pipeline,
-                    RealSense2API.SensorName.COLOR,
-                    RealSense2API.Option.WHITE_BALANCE,
-                    out float min,
-                    out float max,
-                    out float step,
-                    out float def,
-                    out string desc);
+                var option = QueryOption(RealSense2API.Option.WHITE_BALANCE, RealSense2API.SensorName.COLOR);
 
 
                 // step size for depth white balance is 10
-                float adjusted_value = (float)value;
-                int rounding = (value - (int)min) % (int)step;
-                adjusted_value -= (float)rounding;
-
-                if (rounding > step / 2)
-                    adjusted_value += step;
-
-                if (adjusted_value > max)
-                    adjusted_value -= step;
-                if (adjusted_value < min)
-                    adjusted_value += step;
-
+                float adjusted_value = AdjustValue(option.min, option.max, value, option.step);
                 CheckRangeValid<int>(WhiteBalanceDesc, value, (int)adjusted_value, true);
```

Comment:
You could use the `option` object in the CheckRangeValid method (since you have it already).
Using the ParamDesc here repeats the `QueryOption` call and adds no value, but only delay.